DiscordBot.Database/DatabaseCacheRandomFactProvider.cs

```diff
@@ -0,0 +1,44 @@
+﻿using DiscordBot.Core.Interfaces;
+using DiscordBot.Core.Providers;
+using DiscordBot.Database.Repositories;
+
+
+namespace DiscordBot.Database
+{
+    public class DatabaseCacheRandomFactProvider : IRandomFactProvider
+    {
+        private readonly IRandomFactProvider _cacheFactProvider;
+        private readonly IRandomFactRepository _factRepository;
+
+        public DatabaseCacheRandomFactProvider(IRandomFactRepository responseRepository)
+        {
+            _factRepository = responseRepository;
+            _cacheFactProvider = new CacheRandomFactProvider();
+
+            var dbValues = _factRepository.GetAll().GetAwaiter().GetResult();
+            foreach (var dbValue in dbValues)
+            {
+                _cacheFactProvider.Add(Convert.ToUInt64(dbValue.GuildId), dbValue.Fact);
+            }
+        }
+
+        public void Add(ulong? guildId, string randomFact)
+        {
+            if (randomFact is not null)
+                _factRepository.Insert(guildId, randomFact);
+
+            _cacheFactProvider.Add(guildId, randomFact!);
```

Comment:
1. You are not sure here that `randomFact` is not null.
2. We gave the `nullable` enabled for a reason. If the method doesn't accept `string?` then there should be no need to explicitly check it